### ISSUE TITLE
Getting data as array instead of stdClass

### ISSUE THREAD

**Issue by kschroeer** [zfcampus/zf-rest#114]
- [ ] I was not able to find an [open](https://github.com/zfcampus/zf-rest/issues?q=is%3Aopen) or [closed](https://github.com/zfcampus/zf-rest/issues?q=is%3Aclosed) issue matching what I'm seeing.
 - [ ] This is not a question. (Questions should be asked on [chat](https://zendframework.slack.com/) ([Signup here](https://zendframework-slack.herokuapp.com/)) or our [forums](https://discourse.zendframework.com/).)

Provide a narrative description of what you are trying to accomplish.

### Code to reproduce the issue

```php
```

### Actual results

I'm using zf-rest for building my RESTful web apps. I have the following configurations:

    'zf-rest' => [
      Resource\FeedbackResource::class => [
        'listener' => Resource\FeedbackResource::class,
        'route_name' => 'api/rest/feedback',
        'entity_http_methods' => [
        ],
        'collection_http_methods' => [
          'POST',
        ],
      ],
    ],
    'zf-content-validation' => [
      Resource\FeedbackResource::class => [
        'use_raw_data' => false,
        'allows_only_fields_in_filter' => true,
        'POST' => Resource\FeedbackResource::class . '\\Validator',
      ],
    ],
    'input_filter_specs' => [
      Resource\FeedbackResource::class . '\\Validator' => [
        Resource\FeedbackResource::PARAM_NAME => $inputFilterSpecForStrings,
        Resource\FeedbackResource::PARAM_EMAIL => $inputFilterSpecForStrings,
      ],
    ],

Then I created the resource with the corresponding method:

    class FeedbackResource extends AbstractResourceListener
    {
      public function create($data)
      {
        // do something
      }
    }

I posted a json string to the endpoint and everything works fine so far. But what I'm wondering about is that I will get $data as an object with the json data as attributes. I expected to get an assoziative array. Is this possible?

**Comment by alextartan** [comment]
you can always cast it to an array. like: `$dataArray = (array)$data;`

**Comment by TomHAnderson** [comment]
You can pass an object around and change the properties.  An array (not ArrayObject) cannot do that and I believe that's why $data is an object.

**Comment by kschroeer** [comment]
Yes, I know I can use (array), get_object_vars or json_encode/json_decode, I was only wondering about the fact that fetchAll() always gets an array (GET request) and create() gets an object instead (POST request). That seems a bit unlogic.

**Comment by weierophinney** [comment]
> I was only wondering about the fact that fetchAll() always gets an array (GET request) and create() gets an object instead (POST request). That seems a bit unlogic.

How so? A GET request has no request body, so any parameters passed are _query string arguments_; these generally do not map to an object, but are rather considered key/value pairs that typically are not related. As such, we do not map them to objects.

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas-api-tools/api-tools-rest; a new issue has been opened at https://github.com/laminas-api-tools/api-tools-rest/issues/1.
